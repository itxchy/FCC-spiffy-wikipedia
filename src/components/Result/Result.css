.result-cell {
  min-height: 12rem;
  margin-bottom: .2rem;
  padding: .2rem 2rem 1.6rem;
  transition: ease .06s; }
  .result-cell:hover {
    background: #ecf4f9; }

.result-headline {
  text-align: center; }

.result-description {
  text-align: left;
  font-size: 1.2em; }
