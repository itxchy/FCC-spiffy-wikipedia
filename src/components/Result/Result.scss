.result-cell {
  min-height: 12rem;
  margin-bottom: .2rem;
  padding: .2rem 2rem 1.6rem;
  transition: ease .05s;
  &:hover {
    background: lighten(#d8e9f3, 5%);
  }
}