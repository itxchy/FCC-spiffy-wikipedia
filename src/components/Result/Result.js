import React from 'react'
import './Result.css'
const { string } = React.PropTypes

function Result (props) {
  return (
    <article className="result-cell">
      <a href={this.props.link}>
        <h2>{this.props.headline}</h2>
      </a>
      <p>{this.props.description}</p>
    </article>
  )
}

Result.propTypes = {
  link: string,
  headline: string,
  description: string
}

export default Result
