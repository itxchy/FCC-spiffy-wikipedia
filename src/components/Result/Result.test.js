import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Result from './Result'

test('Result snapshot test', () => {
  const component = shallow(<Result link='https://test.test' headline='test' description='testing' />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})
