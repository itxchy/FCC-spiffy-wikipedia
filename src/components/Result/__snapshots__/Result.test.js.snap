exports[`test Result snapshot test 1`] = `
<article
  className="result-cell">
  <a
    href="https://test.test">
    <h2
      className="result-headline">
      test
    </h2>
  </a>
  <p
    className="result-description">
    testing
  </p>
</article>
`;
